### ISSUE TITLE
When logging in, later players may not be able to see who is around them.

### ISSUE THREAD

**Issue by shoshoi** [kevinshen56714/SkyOffice#47]
Hello, thanks for this useful software !

When logging in, later players may not be able to see who is around them. Still, the call can be made.
Sometimes this happens, sometimes it does not.
I am trying this on the same host, with the same browser, and multiple tabs open.

**Comment by shoshoi** [comment]
The earlier report was in error.
When logging in, players who arrive later may not be able to see who is around and cannot make calls.
Seems to happen when the server is under low load.

**Comment by pome-soft** [comment]
It looks like that. But try to login sevral times, the charactors, hided, appear suddenly...

I mean... I have no idea what problem is ...haha..

Do you know any version working right??

**Comment by gusdnvkfks** [comment]
Yeah my problem is like that too.

But I tried sevral times and the characters, hided, appeared suddenly.

I mean I have no idea what the problem is.

plz let me know if you have any idea the software working right way...

**Comment by hawkcookie** [comment]
@gusdnvkfks @shoshoi 
Yes. I have the same problem. Well..,  the reason may be the registration timing of phaserEvents. Especially, ` phaserEvents.emit(Event.PLAYER_JOINED)`  is called before `phaserEvents.on(Event.PLAYER_JOINED)`

**Comment by vinkent420** [comment]
Yeah my problem is like that too with still latest version of Colyseus (0.15)

@kevinshen56714 , do you have any ideas for fixing this one? Thank and so appreciate!

**Comment by droncaglio** [comment]
As @hawkcookie described, if the server is responding to fast, the phaserEvents.emit(Event.PLAYER_JOINED) happens before our player register phaserEvents.on(Event.PLAYER_JOINED). What I did to fix is not elegant, but it works. If someone with more experience on web development can suggest a way to await the event. I fixed by delaying the "emit" like this:

Network.ts

```
initialize(){

// bunch of code



// track changes on every child object inside the players MapSchema
      player.onChange = (changes) => {
        changes.forEach((change) => {
          const { field, value } = change
          phaserEvents.emit(Event.PLAYER_UPDATED, field, value, key)

          // when a new player finished setting up player name
          if (field === 'name' && value !== '') {
            console.log('Network emit Player Joined:', value)

            // delay 0.5 second 
            setTimeout(() => {
              phaserEvents.emit(Event.PLAYER_JOINED, player, key)
              store.dispatch(setPlayerNameMap({ id: key, name: value }))
              store.dispatch(pushPlayerJoinedMessage(value))
            }
              , 500)

          }
        })
      }

// bunch of code
}
```
